.travis.yml

```diff
@@ -11,17 +14,20 @@ env:
 - PYENV_ROOT="$HOME/.pyenv"
 
 install: |
+  set -e
   if [[ ! -d "$PYENV_ROOT/versions" ]]; then
     rmdir "$PYENV_ROOT"
     curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+    git clone https://github.com/momo-lab/xxenv-latest.git "$(pyenv root)/plugins/xxenv-latest"
   fi
   export PATH="$PYENV_ROOT/bin:$PATH"
+  eval "$(pyenv init -)"
+  eval "$(pyenv init --path)"
+  eval "$(pyenv virtualenv-init -)"
   pyenv update
-  git clone https://github.com/momo-lab/xxenv-latest.git "$(pyenv root)/plugins/xxenv-latest"
-  for v in {7..9}; do pyenv latest install -s 3.$v; pyenv latest uninstall -f 3.$v; done
-  pyenv latest local 3.{7..9}
-  pip install --upgrade pip
-  pip install poetry tox
+  for v in {7..10}; do pyenv latest install -s 3.$v; pyenv latest uninstall -f 3.$v; done
```

Comment:
I wonder, is there a distro or another way in travis to have all these pythons precompiled? If not, maybe we could pull binaries? This really seems a huge waste of cycles.

Reply:
The config is supposed to be telling Travis to cache the `pyenv` installs:
```
cache:
- $PYENV_ROOT
```
and this
```
for v in {7..10}; do pyenv latest install -s 3.$v; pyenv latest uninstall -f 3.$v; done
```
is supposed to be ensuring that just the latest release is in the cache: https://github.com/momo-lab/xxenv-latest#usage .

After I get this PR merged, I'll investigate why this isn't working, and improve it (somehow).

Reply:
FWIW, builds on the same branch do cache $PYENV_ROOT correctly:
https://app.travis-ci.com/github/getappmap/appmap-python/jobs/584499355#L262 .

The build for the merged PR didn't use the cache though. :( So, there's definitely room for improvement here.

Reply:
Thanks for looking into this!